### DISCUSSION TITLE
How do I do data validation and custom error message?

### DISCUSSION THREAD

**Discussion by gostan99** [supabase/supabase#1482] [Questions] [answered]
Is it possible to do that?

**Comment by ghost** [comment, marked as answer]
As far as I know, you can't send custom error messages. But, you can send custom email for verification or invites. For data validation, it is already handled by [PostgREST API Server](https://supabase.io/docs/postgrest/server/about#performance).

**Comment by silentworks** [comment]
You can handle data validation inside of your own code and also create custom error messages in your own code.
